templates/types/streaming/nextjs/app/components/ui/chat/chat-message/chat-agent-events.tsx

```diff
@@ -136,15 +200,41 @@ function mergeAdjacentEvents(events: AgentEventData[]): MergedEvent[] {
   for (const event of events) {
     const lastMergedEvent = mergedEvents[mergedEvents.length - 1];
 
+    // Check the event_type first
+    if (event.type === "progress") {
+      try {
+        const progressData = event.data;
+
+        if (lastMergedEvent && lastMergedEvent.agent === event.agent) {
+          lastMergedEvent.progress = progressData;
+          lastMergedEvent.texts.push(event.text);
+        } else {
+          mergedEvents.push({
+            agent: event.agent,
+            texts: [],
+            icon: AgentIcons[event.agent.toLowerCase()] ?? icons.Bot,
+            progress: progressData,
+          });
+        }
+      } catch (e) {
+        console.error(
+          "Failed to parse progress data:",
+          e,
+          "Raw text:",
+          event.text,
+        );
+      }
+      continue;
+    }
+
+    // Handle regular text events
     if (lastMergedEvent && lastMergedEvent.agent === event.agent) {
-      // If the last event in mergedEvents has the same non-null agent, add the title to it
       lastMergedEvent.texts.push(event.text);
```

Comment:
Seems way easier to me to just check for progress here and if it exists adding it